examples/runRieszWaveletAnalysis.cpp

```diff
@@ -0,0 +1,396 @@
+  std::string outputFile = AppendToFilenameRiesz(outputImage, appendString);
+  writer->SetFileName( outputFile );
+  writer->SetInput( caster->GetOutput() );
+
+  writer->Update();
+  std::cout << "Output generated in: " << outputFile << std::endl;
+  return EXIT_SUCCESS;
+}
+
+int main( int argc, char *argv[] )
+{
+  if ( argc < 8 || argc > 9 )
+    {
+    std::cerr << "Usage: " << argv[0]
+              << " inputImage outputImage inputLevels inputBands waveletFunction dimension Apply|NoApply [thresholdNumOfSigmas]"
+              << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  const std::string inputImage  = argv[1];
+  const std::string outputImage = argv[2];
+  const unsigned int inputLevels = atoi( argv[3] );
+  const unsigned int inputBands  = atoi( argv[4] );
+  const std::string waveletFunction = argv[5];
+  const unsigned int dimension = atoi( argv[6] );
+  const std::string applySoftThresholdInput = argv[7];
+  bool applySoftThreshold = false;
+  if ( applySoftThresholdInput == "Apply" )
+    {
+    applySoftThreshold = true;
+    }
+  else if ( applySoftThresholdInput == "NoApply" )
+    {
+    applySoftThreshold = false;
+    }
+  else
+    {
+    std::cerr << "Unkown string: " + applySoftThresholdInput + " . Use Apply or NoApply." << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  double thresholdNumOfSigmas = 2.0;
+  if ( argc == 9 )
+    {
+    thresholdNumOfSigmas = atof(argv[8]);
+    }
+
+  constexpr unsigned int ImageDimension = 2;
+  using PixelType = float;
+  using ComplexPixelType = std::complex< PixelType >;
+  using PointType = itk::Point< PixelType, ImageDimension >;
+  using ComplexImageType = itk::Image< ComplexPixelType, ImageDimension >;
+
+  // Exercise basic object methods
+  // Done outside the helper function in the test because GCC is limited
+  // when calling overloaded base class functions.
+  using HeldIsotropicWaveletType = itk::HeldIsotropicWavelet< PixelType, ImageDimension, PointType >;
+  using VowIsotropicWaveletType = itk::VowIsotropicWavelet< PixelType, ImageDimension, PointType >;
+  using SimoncelliIsotropicWaveletType = itk::SimoncelliIsotropicWavelet< PixelType, ImageDimension, PointType >;
+  using ShannonIsotropicWaveletType = itk::ShannonIsotropicWavelet< PixelType, ImageDimension, PointType >;
+
+  auto heldIsotropicWavelet =
+    HeldIsotropicWaveletType::New();
+
+  auto vowIsotropicWavelet =
+    VowIsotropicWaveletType::New();
+
+  auto simoncellidIsotropicWavelet =
+    SimoncelliIsotropicWaveletType::New();
+
+  auto shannonIsotropicWavelet = ShannonIsotropicWaveletType::New();
+
+  using HeldWavelet = itk::HeldIsotropicWavelet< >;
+  using VowWavelet = itk::VowIsotropicWavelet< >;
+  using SimoncelliWavelet = itk::SimoncelliIsotropicWavelet< >;
+  using ShannonWavelet = itk::ShannonIsotropicWavelet< >;
+
+  using HeldWaveletFilterBankType = itk::WaveletFrequencyFilterBankGenerator< ComplexImageType, HeldWavelet >;
```

Comment:
Types and variables on lines 271-298 don't seem to be used. If you just wanted to exercise all of that, it is better to put it in a unit test which does just that, and not anything ordinarily useful. Keep the useful stuff in the example. For unit tests, you might want to consider [EXERCISE_BASIC_OBJECT_METHODS](https://github.com/InsightSoftwareConsortium/ITK/blob/c95b99852a9109a8fc176a06f88b2aa6d442154b/Modules/Video/IO/test/itkVideoFileReaderWriterTest.cxx#L53) and other stuff from [itkTestingMacros.h](https://github.com/InsightSoftwareConsortium/ITK/blob/c95b99852a9109a8fc176a06f88b2aa6d442154b/Modules/Core/TestKernel/include/itkTestingMacros.h#L60)

Reply:
Thanks, this example was extracted from a unit test, so some leftovers were unnecessarily left here.

Reply:
Removed in last commit.